crates/typst/src/layout/pages/run.rs

```diff
@@ -122,6 +123,10 @@ fn layout_page_run_impl(
     let header_ascent = PageElem::header_ascent_in(styles).relative_to(margin.top);
     let footer_descent = PageElem::footer_descent_in(styles).relative_to(margin.bottom);
     let numbering = PageElem::numbering_in(styles);
+    let supplement = match PageElem::supplement_in(styles) {
+        Smart::Auto => TextElem::packed(PageElem::local_name_in(styles)),
+        Smart::Custom(content) => content.unwrap_or_default(),
+    };
```

Comment:
I only just noticed while seeing this that referencing a page in a run where `set text(lang: "de")` from a page run where `set text(lang: "en")` will render `as we can see on Seite 10`. I'm not sure how much sense that makes. But it's similar with headings and stuff ...

Reply:
It's the same as this issue, I think: https://forum.typst.app/t/how-to-have-different-supplements-for-the-same-heading-depending-on-the-location/667

Reply:
I can see how someone might find it odd either way. I think it'd be best then to keep this consistent with headings and all for now.

Reply:
Also, this same behaviour (but with setting the supplement instead of setting languages) is shown in the `ref-form-page` test, I believe.